lib/govuk_publishing_components/presenters/breadcrumb_selector.rb

```diff
@@ -45,11 +45,16 @@ def options(navigation)
             step_by_step: false,
             breadcrumbs: navigation.taxon_breadcrumbs,
           }
-        elsif navigation.content_tagged_to_mainstream_browse_pages?
+        elsif navigation.content_tagged_to_mainstream_browse_pages? && navigation.content_parent_is_a_mainstream_browse_page?
```

Comment:
What do you think about just having one method in ContextualNavigation called `content_parent_is_mainstream_browse?`. As that's all the breadcrumb selector needs right? It's not using the fact that it's tagged to mainstream browse.

Reply:
So changing this check actually breaks the logic somehow and causes a failure where the breadcrumbs are incorrect (text expecting taxons rather than parent). Investigating. There are definitely cases set up in the test environment where the parent is an MSB but there is no content tagged, so I broke Taxon rendering in that case by removing the check that it's tagged. Oops.

Reply:
I attempted this but it appears doing so breaks cases where a page has a MSB parent but no MSB tags. Flagged by https://github.com/alphagov/govuk_publishing_components/blob/add-mainstream-topics-to-breadcrumbs/spec/components/contextual_breadcrumbs_spec.rb#L96

This flags that there are cases where a we don't want to render the parent if it exists, specifically when there are now MSB related content tagged to the page. I'm hesitant to rework this behaviour at this point as I am unsure on the splash it will have.

I have however renamed the function as your suggestion here is much cleaner.

Reply:
> cases where a page has a MSB parent but no MSB tags

I don't think that is a possible scenario. AFAIK it's only Publisher which allows pages to have a MSB added to the parent field (in the UI it's called `Breadcrumb`), and there's a restriction that the MSB page chosen for the parent field has to match the MSB page add to the MSB page field
<img width="842" alt="Screenshot 2024-02-12 at 12 05 49" src="https://github.com/alphagov/govuk_publishing_components/assets/17908089/6c5c215f-3e42-4a84-bb0d-8f2081dc5b51">

Reply:
So unless I'm missing something, it sounds like the tests contain unrealistic data.

Reply:
Is it possible that it was once valid but no longer is? Either way I've just removed both the redundant check MSB and the invalid test. Now only checks for parent.